### DISCUSSION TITLE
How can I import a Resource of a Project on Digitalocean?

### DISCUSSION THREAD

**Discussion by fuadnafiz98** [digitalocean/terraform-provider-digitalocean#1032] [Q&A]
I am trying to import a `droplet` via `terraform import` but as the droplet is inside a project. I cannot import it. I also looked into the docs and found that the importing of resources of a project is not supported. https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/project_resources

Any Idea how I can retrieve the information of a resource even if its inside a project not made by terraform?

**Comment by andrewsomething** [comment]
Droplets inside a project not created by Terraform should be able to be imported without issue. Though if the Droplet is recreated it will be placed in the default project unless you also use the `digitalocean_project_resources` resource.

The `digitalocean_project_resources` resource does not need to manage every resource in the project. It can be used to manage a subset of them. So after importing the Droplet, you could just add something like:

```
resource "digitalocean_project_resources" "foo" {
  project = "aaa-bbb-ccc"
  resources = [
    digitalocean_droplet.foo.urn
  ]
}

resource "digitalocean_droplet" "foo" {
  name   = "foo"
  image  = "123456"
  region = "nyc3"
  size   = "s-1vcpu-1gb"
  monitoring = true
}
```

This will "create" a new `digitalocean_project_resources` resource, but as the Droplet is already in the project there is no functional change.

**Comment by fuadnafiz98** [comment]
Thank you for the comment. 

But I am not been able to import the droplet too. 

When I try to import it via terraform import, it doesn’t show anything on the output. 

For instance, I can able to import a project and it shows the terraform state of the project, but I couldn’t able to done the same for droplets. 

My assumptions was that due to the existence of the droplet is inside a project, it couldn’t able to find it.

**Comment by andrewsomething** [comment]
Can you share what you are seeing when you attempt to import the Droplet?

**Comment by fuadnafiz98** [comment]
Do you want the info with trace logs? 

It shows that the resource has imported successfully, but nothing shows on the state. The output is empty.

**Comment by fuadnafiz98** [comment]
Hi, 
Can you please have a look on the output here. 

![image](https://github.com/digitalocean/terraform-provider-digitalocean/assets/43469396/0b704036-8821-470a-a08a-6c06cd12b9bf)

You can see it shows that the import is successful, but there is noting imported shown above.

**Comment by fuadnafiz98** [comment]
FYI 
I am using this command: 

```
terraform import -var "do_token=${DO_TOKEN}" digitalocean_droplet.droplet_import <droplet_id>

```
And this is on my `droplet_import.tf` file

```
resource "digitalocean_droplet" "droplet_import" {
    name   = "server-name"
    region = "sgp1"
    tags   = []
    count  = "1"
}

```

**Comment by fuadnafiz98** [comment]
Hi, I have fixed the issue. 

I needed to write the `droplet_import.tf` like this

```
resource "digitalocean_droplet" "droplet_import" {
    name   = "server-name"
}
```

And it was imported successfully.
